File: BenchmarkProblems/algorithms/costMatrixRepeated.m
function C = costMatrixRepeated(E1, E2)
    n = length(E1);
    C = zeros(n);
    
    % sort rows of E1 and E2
    for i = 1:n
        E1(i, :) = sort(E1(i, :));
        E2(i, :) = sort(E2(i, :));
    end
    
    % compare sorted vectors
    for i = 1:n
        for j = 1:n
            C(i, j) = norm(E1(i, :) - E2(j, :));
        end
    end
end

File: GI_benchmark/costMatrixRepeated.jl
function costMatrixRepeated(E1::AbstractMatrix, E2::AbstractMatrix)
    n = size(E1, 1)
    C = spzeros(Float64, n, n)

    # sort rows of E1 and E2
    E1s = [sort(E1[i, :]) for i in 1:n]
    E2s = [sort(E2[i, :]) for i in 1:n]

    # compare sorted vectors
    for i in 1:n
        for j in 1:n
            C[i, j] = norm(E1s[i] .- E2s[j])
        end
    end

    return C
end
